src/pages/index.js

```diff
@@ -29,50 +29,165 @@ import {
   previewFooter,
   formValidationConfig,
 } from "../utils/constants.js";
+import Api from "../components/Api.js";
+import PopupWithConfirmation from "../components/PopupWithConfirmation.js";
 
-// New Card
+// =============================================================================
+// API + userInfo
+// =============================================================================
 
-function createCard({ name, link }) {
+let userID;
+let cardList;
+
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "a9aaf24d-57aa-44df-b0d5-364cf1f03985",
+    "Content-Type": "application/json",
+  },
+});
+
+const userInfo = new UserInfo({
+  nameSelector: ".profile__title",
+  professionSelector: ".profile__description",
+  avatarSelector: ".profile__image",
+});
+
+Promise.all([api.getUserInfo(), api.getInitialCards()]).then(
+  ([user, initialCards]) => {
+    userInfo.setUserInfo(user);
+    userID = user._id;
+    userInfo.setUserInfo(user);
+
+    cardList = new Section(
+      {
+        items: initialCards,
+        renderer: ({ name, link, isLiked, _id, userId, ownerId }) => {
+          const newCard = createCard({
+            name,
+            link,
+            isLiked,
+            _id,
+            userId,
+            ownerId,
+          });
+          cardList.addItem(newCard);
+        },
+      },
+      ".cards__list"
+    );
+
+    cardList.renderItems();
+  }
+);
+// =============================================================================
+// Handle Clicks
+// =============================================================================
+
+function handleLikeClick() {
+  if (this._likes) {
+    api
+      .unlikeCard(this._id)
+      .then((res) => {
+        this.setLikes(res.isLiked);
```

Comment:
`this` should **not** be used outside classes in functions in `index.js` . This context is easy to lose. You need to find other ways of passing data.